.github/workflows/linux-build-and-test.yaml

```diff
@@ -0,0 +1,143 @@
+name: Regression Linux
+on:
+  schedule:
+    # run daily 20:00 on master branch
+    - cron: '0 20 * * *'
+  push:
+    branches:
+      - prerelease_test
+  pull_request:
+jobs:
+  regress_linux:
+    name: PG${{ matrix.pg }} [${{ matrix.type }}]
+    runs-on: ubuntu-18.04
+    strategy:
+
+# This is the main workflow for linux regression tests
+# Since GitHub Actions don't currently allow dynamic
+# matrix entries, the matrix has all versions but
+# for pull request we only run the configurations
+# where on_pullrequest is true
+
+      matrix:
+        pg: [ "11.0", "11.8", "12.0", "12.3" ]
+        type: [ Debug, Release, ApacheOnly ]
+
+        include:
+          - type: Debug
+            on_pullrequest: true
+            build_type: Debug
+            pg_build_args: --enable-debug --enable-cassert
+            tsdb_build_args: -DCODECOVERAGE=ON
+            coverage: true
+          - type: Release
+            build_type: Release
+          - type: ApacheOnly
+            build_type: Release
+            tsdb_build_args: -DAPACHE_ONLY
+          - pg: "11.0"
+            on_pullrequest: false
+          - pg: "12.0"
+            on_pullrequest: false
+        exclude:
+          - pg: "11.0"
+            type: ApacheOnly
+          - pg: "11.0"
+            type: Release
+          - pg: "12.0"
+            type: ApacheOnly
+          - pg: "12.0"
+            type: Release
+    env:
+      PG_SRC_DIR: /home/runner/postgresql
+    steps:
+
+    - name: Install Dependencies
+      if: matrix.on_pullrequest || github.event_name != 'pull_request'
+      run: sudo apt install lcov
+
+    - name: Cache PostgreSQL
+      if: matrix.on_pullrequest || github.event_name != 'pull_request'
+      id: cache-postgresql
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.PG_SRC_DIR }}
+        key: ${{ runner.os }}-postgresql-${{ matrix.pg }}-${{ matrix.build_type }}
+
+    - name: Build PostgreSQL ${{ matrix.pg }}
+      if: ( matrix.on_pullrequest || github.event_name != 'pull_request' ) && steps.cache-postgresql.outputs.cache-hit != 'true'
+      run: |
+        wget -q -O postgresql.tar.bz2 https://ftp.postgresql.org/pub/source/v${{ matrix.pg }}/postgresql-${{ matrix.pg }}.tar.bz2
+        mkdir -p $PG_SRC_DIR
+        tar --extract --file postgresql.tar.bz2 --directory $PG_SRC_DIR --strip-components 1
+        cd $PG_SRC_DIR
+        ./configure --prefix=/usr/local ${{ matrix.pg_build_args }} --with-openssl --with-llvm --without-readline --without-zlib
```

Comment:
Why put this in `/usr/local` instead of a subdir of workdir? I guess this would unnecessarily require sudo for a bunch of install steps below

Reply:
This wasn't answered...

Reply:
main reason was so the binaries will end up in PATH but this might actually not be needed i'll check